core/src/main/java/io/fabric8/maven/core/util/ProfileUtil.java

```diff
@@ -71,6 +74,18 @@ public static Profile findProfile(String profileArg, File resourceDir) throws IO
         }
     }
 
+    private static Profile inheritFromParentProfile(Profile aProfile, File resourceDir) throws IOException {
+        Profile aParentProfile = lookup(aProfile.getParentProfile(), resourceDir);
+        if(aParentProfile != null) {
+            aProfile.setEnricherConfig(ProcessorConfig.mergeProcessorConfigs(aProfile.getEnricherConfig(), aParentProfile.getEnricherConfig()));
+            aProfile.setGeneratorConfig(ProcessorConfig.mergeProcessorConfigs(aProfile.getGeneratorConfig(), aParentProfile.getGeneratorConfig()));
+            aProfile.setWatcherConfig(ProcessorConfig.mergeProcessorConfigs(aProfile.getWatcherConfig(), aParentProfile.getWatcherConfig()));
+        } else {
+            throw new IllegalArgumentException("No profile '" + aProfile.getParentProfile() + "' defined");
```

Comment:
It would be better to log "No parent profile xyz defined". Let it be explicit for user. WDYT?